src/wp-includes/class-wp-block-type.php

```diff
@@ -613,7 +639,41 @@ public function get_variations() {
 		 * @param array         $variations Array of registered variations for a block type.
 		 * @param WP_Block_Type $block_type The full block type object.
 		 */
-		return apply_filters( 'get_block_type_variations', $this->variations, $this );
+		$variations = apply_filters( 'get_block_type_variations', $this->variations, $this );
+
+		foreach ( $variations as $variation ) {
+			if ( ! empty( $variation['supports']['alias'] ) && true === $variation['supports']['alias'] ) {
+				$this->aliases[ $variation['name'] ] = $variation;
+			}
+		}
+		return $variations;
+	}
+
+	/**
+	 * Get block aliases.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @return array[]
+	 */
+	public function get_aliases() {
+		if ( ! isset( $this->aliases ) ) {
+			$this->get_variations();
+
+			if ( ! isset( $this->aliases ) ) {
+				$this->aliases = array();
+			}
+		}
+
+		/**
+		 * Filters the registered aliases for a block type.
+		 *
+		 * @since 6.6.0
+		 *
+		 * @param array         $aliases    Array of registered aliases for a block type.
+		 * @param WP_Block_Type $block_type The full block type object.
+		 */
+		return apply_filters( 'get_block_type_aliases', $this->aliases, $this );
```

Comment:
I know this already matches the existing pattern of using a filter similar to `get_block_type_variations`, but it's always felt odd to me that there are not more declarative APIs to register variations or in this case, aliases.

Reply:
There is a method that is a bit more declarative, you can use the `variation_callback` when registering the block itself which you can see an example of [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/post-terms/index.php#L70).

It still feels different to the client-side registration where you can register them separately using an imported function.